packages/artifact/src/internal/client.ts

```diff
@@ -225,4 +244,41 @@ If the error persists, please check whether Actions and API requests are operati
       throw error
     }
   }
+
+  async deleteArtifact(
+    artifactName: string,
+    options?: FindOptions
+  ): Promise<DeleteArtifactResponse> {
+    try {
+      if (isGhes()) {
+        throw new GHESNotSupportedError()
+      }
+
+      if (options?.findBy) {
+        const {
+          findBy: {repositoryOwner, repositoryName, workflowRunId, token}
+        } = options
+
+        return deleteArtifactPublic(
+          artifactName,
+          workflowRunId,
+          repositoryOwner,
+          repositoryName,
+          token
+        )
+      }
+
+      return deleteArtifactInternal(artifactName)
+    } catch (error) {
+      warning(
```

Comment:
Since there is no `await` for `deleteArtifactPublic` / `deleteArtifactInternal` is this warning actually printed when something goes wrong asynchronously? To me it looks this only catches errors which are thrown synchronously.